### PR TITLE
feat: support None operand in EQUAL operator

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
This PR intends to support the `None` operand with `Equal` and `Not-Equal` operators in Filters in QueryObject. Currently, the `In` operator has supported `None` operand calculation, in order to align that the `Equal` and` Not-Equal` should support as well.
 
After the PR, there are 4 approaches to generate `is null` or `is not null` filter in SQL.

1. `EQUAL` / `NOT EQUAL` operator
```Python
>>> result = mock_dataset.query({
    "metrics": ["count"],
    "filter": [{"col": "col", "val": None, "op": "=="}],
    "is_timeseries": False,
})
>>> result.query
SELECT count(*) AS count
FROM mock_dataset
WHERE col IS NULL
```

2. `IS NULL` / `IS NOT NULL` operator
```Python
>>> result = mock_dataset.query({
    "metrics": ["count"],
    "filter": [{"col": "col", "op": "IS NULL"}],
    "is_timeseries": False,
})
>>> result.query
SELECT count(*) AS count
FROM mock_dataset
WHERE col IS NULL
```

3. `IN` operator
```Python
>>> result = mock_dataset.query({
    "metrics": ["count"],
    "filter": [{"col": "col", val: ["a", None], "op": "IN"}],
    "is_timeseries": False,
})
>>> result.query
SELECT count(*) AS count
FROM mock_dataset
WHERE col IS NULL
  OR col IN ('a')
```

4. There is an additional case that `<NULL>` as filter operand literal constant in the EQ or IN operators
```Python
>>> result = mock_dataset.query({
    "metrics": ["count"],
    "filter": [{"col": "col", "val": '<NULL>', "op": "=="}],
    "is_timeseries": False,
})
>>> result.query
SELECT count(*) AS count
FROM mock_dataset
WHERE col IS NULL
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/integration_tests/sqla_models_tests.py [hunk 1]
```diff
@@ -659,51 +660,90 @@ def test_filter_on_text_column(text_column_table):
...
-        TableColumn(
-            column_name="datetime_col",
-            type="TIMESTAMP",
-            table=table,
-            is_dttm=True,
-        )
-        SqlMetric(metric_name="count", expression="count(*)", table=table)
-        result_object = table.query(
+@only_postgresql
+def test_should_generate_closed_and_open_time_filter_range(login_as_admin):
```
Comment (zhaoyongjie) [inline tests/integration_tests/sqla_models_tests.py:664]: bycatch: use `only_postgresql` decorator narrowing test case, and use `login_as_admin` to define the Flask app context.

File: tests/integration_tests/sqla_models_tests.py [hunk 2]
```diff
@@ -659,51 +660,90 @@ def test_filter_on_text_column(text_column_table):
...
+               UNION SELECT '2023-01-01'::timestamp
+               UNION SELECT '2023-03-10'::timestamp) AS virtual_table
+            WHERE datetime_col >= TO_TIMESTAMP('2022-01-01 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US')
+              AND datetime_col < TO_TIMESTAMP('2023-01-01 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US')
+    """
+    assert result_object.df.iloc[0]["count"] == 2
+
+
+def test_none_operand_in_filter(login_as_admin, physical_dataset):
+    expected_results = [
```
Comment (zhaoyongjie) [inline tests/integration_tests/sqla_models_tests.py:708]: This test case is the new one.

**Review by villebro — COMMENTED** [review]
I hadn't noticed it actually allowed selecting the empty value despite the API not allowing it 😵‍💫 Works great! However, one bug that I noticed when testing + a few stylistic nits:

1. When creating a new filter, then selecting "IS NOT NULL" operator and then switching back to "equals" or "not equals", the save button becomes disabled: 
image

2. We should also update the pill labels to reflect that they are checking for _NULL_. For the "equals" case I'd do something like "col = NULL" and "col <> NULL":
image
image
3. Also a bycatch, but I'd actually prefer for the NOT EQUALS operator to use the same ≠ character in the pill label that we're using in the dropdown:
image

**Review by villebro — APPROVED** [review]
LGTM - leaving my comments on the table for a potential follow-up PR as they're mostly unrelated

**Comment by zhaoyongjie** [comment]
@villebro Thanks a lot, I will fix the frontend problem in the separated PR.

**Comment by john-bodley** [comment]
@zhaoyongjie did you address the frontend issues that @villebro mentioned in a separate PR? If so could you reference this PR in said PR so there's a edge connecting the two PRs?

**Comment by zhaoyongjie** [comment]
@john-bodley I will do these cases. thanks for the mention.

**Pull request merged**